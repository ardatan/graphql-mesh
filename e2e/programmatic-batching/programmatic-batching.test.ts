import { createTenv } from '@e2e/tenv';

const { compose, service, serve } = createTenv(__dirname);

it('should compose the appropriate schema', async () => {
  const { result } = await compose({
    services: [await service('api')],
    maskServicePorts: true,
  });
  expect(result).toMatchSnapshot();
});

it.concurrent.each([
  {
    name: 'User',
    query: /* GraphQL */ `
      fragment UserF on User {
        id
        name
      }
      query User {
        john: user(id: 1) {
          ...UserF
        }
        jane: user(id: 2) {
          ...UserF
        }
      }
    `,
  },
])('should execute $name', async ({ query }) => {
  const { output } = await compose({ output: 'graphql', services: [await service('api')] });
  const { execute } = await serve({ supergraph: output });
  await expect(execute({ query })).resolves.toMatchSnapshot();
});
