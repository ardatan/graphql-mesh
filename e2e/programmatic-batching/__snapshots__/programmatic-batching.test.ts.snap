// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should compose the appropriate schema 1`] = `
"schema @transport(subgraph: "API", kind: "rest", location: "http://0.0.0.0:<api_port>") {
  query: Query
  mutation: Mutation
}

directive @httpOperation(subgraph: String, path: String, operationSpecificHeaders: ObjMap, httpMethod: HTTPMethod, isBinary: Boolean, requestBaseBody: ObjMap, queryParamArgMap: ObjMap, queryStringOptionsByParam: ObjMap, jsonApiFields: Boolean) on FIELD_DEFINITION

directive @transport(subgraph: String, kind: String, location: String, headers: ObjMap, queryStringOptions: ObjMap, queryParams: ObjMap) on OBJECT

type Query {
  dummy: String @source(subgraph: "API", name: "dummy", type: "String")
  user(id: Float!): User
}

type Mutation {
  usersByIds(input: UsersByIdRequest_Input): UsersByIdResponse @httpOperation(subgraph: "API", path: "/users_by_ids", operationSpecificHeaders: "\\"{\\\\\\"Content-Type\\\\\\":\\\\\\"application/json\\\\\\",\\\\\\"accept\\\\\\":\\\\\\"application/json\\\\\\"}\\"", httpMethod: POST) @source(subgraph: "API", name: "usersByIds", type: "UsersByIdResponse")
}

type UsersByIdResponse @source(subgraph: "API", name: "UsersByIdResponse") {
  results: [User]! @source(subgraph: "API", name: "results", type: "[User]!")
}

type User @source(subgraph: "API", name: "User") {
  id: Float! @source(subgraph: "API", name: "id", type: "Float!")
  name: String! @source(subgraph: "API", name: "name", type: "String!")
}

input UsersByIdRequest_Input @source(subgraph: "API", name: "UsersByIdRequest_Input") {
  ids: [Float]! @source(subgraph: "API", name: "ids", type: "[Float]!")
}

scalar ObjMap @source(subgraph: "API", name: "ObjMap") @source(subgraph: "API", name: "ObjMap")

enum HTTPMethod @source(subgraph: "API", name: "HTTPMethod") {
  GET @source(subgraph: "API", name: "GET")
  HEAD @source(subgraph: "API", name: "HEAD")
  POST @source(subgraph: "API", name: "POST")
  PUT @source(subgraph: "API", name: "PUT")
  DELETE @source(subgraph: "API", name: "DELETE")
  CONNECT @source(subgraph: "API", name: "CONNECT")
  OPTIONS @source(subgraph: "API", name: "OPTIONS")
  TRACE @source(subgraph: "API", name: "TRACE")
  PATCH @source(subgraph: "API", name: "PATCH")
}
"
`;

exports[`should execute User 1`] = `
{
  "data": {
    "jane": {
      "id": 2,
      "name": "Jane Doe",
    },
    "john": {
      "id": 1,
      "name": "John Doe",
    },
  },
}
`;
